### PR TITLE
Add bsp_pin example

### LINKED ISSUE #262: Explain the bsp_pins macro
Reading https://github.com/rp-rs/rp-hal/pull/259/files made me realise that the bsp_pins macro could probably do with some explanatory text. Like,

```
/// Calling the macro like this:
/// ```
/// bsp_pins! {
///    Gpio0 { name: gpio0 },
/// }
/// ```
/// 
/// Is equivalent to the following source code:
/// ```
/// <I have no idea what goes here>
/// ```
```

### PR DESCRIPTION
**Pull request by hmvp**
Fixes #262

### CODE REVIEW

File: rp2040-hal/src/gpio/pin.rs [hunk 1]
```diff
@@ -1075,7 +1075,55 @@ gpio!(
...
+/// }
+/// ```
+///
+/// Is roughly equivalent to the following source code (excluding the docs strings below):
+/// ```
+/// use ::rp2040_hal as hal;
+/// use hal::gpio;
+/// pub struct Pins {
+///     # [cfg (feature != "gpio")]
+///     /// Doc gpio0"]
```
Comment (jannic) [inline rp2040-hal/src/gpio/pin.rs:1100]: The last two chars, `"]`, look a little bit weird. Is this intended / correct?

File: rp2040-hal/src/gpio/pin.rs [hunk 2]
```diff
@@ -1075,7 +1075,55 @@ gpio!(
...
+///      },
+/// }
+/// ```
+///
+/// Is roughly equivalent to the following source code (excluding the docs strings below):
+/// ```
+/// use ::rp2040_hal as hal;
+/// use hal::gpio;
+/// pub struct Pins {
+///     # [cfg (feature != "gpio")]
```
Comment (jannic) [inline rp2040-hal/src/gpio/pin.rs:1099]: `#[cfg(not(feature = "gpio"))]` ?

Comment (hmvp) [reply rp2040-hal/src/gpio/pin.rs]: Yeah some hacky workaround, I removed it already

**Comment by 9names** [comment]
Looks good, thanks!

**Pull request merged**